packages/block-editor/src/components/block-tools/index.js

```diff
@@ -59,19 +87,18 @@ export default function BlockTools( {
 	__unstableContentRef,
 	...props
 } ) {
-	const { clientId, hasFixedToolbar, isTyping, isZoomOutMode } = useSelect(
-		selector,
-		[]
-	);
-	const isMatch = useShortcutEventMatch();
-	const { getSelectedBlockClientIds, getBlockRootClientId } =
-		useSelect( blockEditorStore );
```

Comment:
The static selector getter doesn't create a store subscription. It's fine to have this separately.

Reply:
Can you explain this more? Performance around `useSelect` is an area I would like to get better at understanding.

Reply:
Store subscriptions can add up and affect performance, especially for active stores like `core/block-editor`. If the selector is only used in event callback or in effect, then it's better to use a static getter that doesn't create subscriptions.

In this case, removing this like and returning selectors from the `mapSelect` doesn't make a difference, and it's fine to have this getter.

```js
// Doesn't create a sub.
const { getBlockRootClientId } = useSelect( blockEditorStore );

// Does create a sub.
const { getBlockRootClientId } = useSelect( ( select ) => select( blockEditorStore ), [] );
```

P.S. Jarda recently wrote a great article about this subject - https://developer.wordpress.org/news/2024/03/28/how-to-work-effectively-with-the-useselect-hook/.